task-tracker-console-application/src/main/java/com/isa/user/Login.java

```diff
@@ -0,0 +1,135 @@
+package com.isa.user;
+
+import com.isa.menu.MainMenu;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Scanner;
+
+public class Login {
+    private static User loggedInUser;
+    private static final List<User> users = User.getTestUsers();
+    private static String inputLogin;
+
+    public static User getLoggedInUser() {
+        return loggedInUser;
+    }
+
+    public static void logOutUser() {
+        loggedInUser = null;
+    }
+
+    private static void displayLoginMessage() {
+        System.out.println("=====================================");
+        System.out.println("             ZALOGUJ SIĘ");
+        System.out.println("=====================================");
+        System.out.println("Podaj dane logowania lub wybierz \"0\", żeby powrócić do menu głównego.");
+    }
+
+    private static void logInUser(Scanner scanner) {
+        System.out.print("Podaj login: ");
+        validateLogin(scanner, 0);
+    }
+
+    private static void validateLogin(Scanner scanner, int incorrectLoginInputCount) {
+        inputLogin = scanner.nextLine();
+        if (inputLogin.equals("0")) {
+            goToMainMenu();
+        } else if (chosenUser().isEmpty()) {
+            incorrectLoginInputCount++;
+            System.out.println("Podany login nie istnieje.");
+            if (incorrectLoginInputCount > 1) {
+                System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+            }
+
+            System.out.print("Podaj login ponownie: ");
+            validateLogin(scanner, incorrectLoginInputCount);
+        } else {
+            System.out.print("Podaj hasło: ");
+            validatePassword(scanner, 0);
+        }
+    }
+
+    private static Optional<User> chosenUser() {
+        return Optional.ofNullable(users.stream().
+                filter(user -> user.getLogin().equals(inputLogin)).findFirst().orElse(null));
+    }
```

Comment:
tutaj można odpakować to z tego optionala, bo metoda `findFirst()` zwraca nam optionala, dodatkowo w optionalach czy streamach dobrą praktyką jest pisanie od nowej linii każdego etapu, który się rozpoczyna kropką, np.
```
private static Optional<User> chosenUser() {
    return users.stream()
        .filter(user -> user.getLogin().equals(inputLogin))
        .findFirst();
}
```

Reply:
Poprawione zgodnie z sugestią; dzięki!